core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java

```diff
@@ -492,13 +502,34 @@ void createMeasurementEvent(
             Unsafe.getUnsafe().putLong(bufPos, timestamp);
             bufPos += Long.BYTES;
             int nEntities = protoParser.getnEntities();
-            Unsafe.getUnsafe().putInt(bufPos, nEntities);
+
+            final long entitiesBufPos = bufPos;
+            Unsafe.getUnsafe().putInt(entitiesBufPos, nEntities);
             bufPos += Integer.BYTES;
+
+            final IntList currentAccessCounters = localDetails.accessCountersByColumnIndex;
+            final LowerCaseCharSequenceIntHashMap accessCountersByColumnName =
+                    localDetails.accessCountersByColumnName;
+
+            final int countersSize = currentAccessCounters.size();
+            for (int i = 0; i < countersSize; i++) {
+                currentAccessCounters.setQuick(i, 0);
+            }
+
+            int writtenEntities = 0;
+            boolean undefinedColumnEncountered = false;
+
             for (int nEntity = 0; nEntity < nEntities; nEntity++) {
                 if (bufPos + Long.BYTES < bufMax) {
                     ProtoEntity entity = protoParser.getEntity(nEntity);
                     int colIndex = localDetails.getColumnIndex(entity.getName());
+
                     if (colIndex < 0) {
+                        if (!undefinedColumnEncountered) {
+                            accessCountersByColumnName.updateAllValues(0);
```

Comment:
This will leave all keys in place. Wouldn't `accessCountersByColumnName.clear()` be better?

Reply:
Yep, I mentioned in the explanation. I will change it to clear. :-)